openedx/core/djangoapps/content_tagging/rest_api/v1/tests/test_views.py

```diff
@@ -142,105 +214,165 @@ def setUp(self):
+        "content_creatorA",
+        "instructorA",
+        "library_staffA",
+        "course_instructorA",
+        "course_staffA",
+        "library_userA",
+    )
+    def test_list_taxonomy_orgA(self, user_attr: str) -> None:
+        """
+        Tests that non staff users from orgA can see only enabled taxonomies from orgA and global taxonomies
+        """
+        expected_taxonomies = ["st1", "t1", "tA1", "tBA1"]
+        self._test_list_taxonomy(
+            user_attr=user_attr,
+            enabled_parameter=True,
+            expected_taxonomies=expected_taxonomies,
+        )
+
+    @ddt.data(
+        (True, ["ot1", "st1", "t1", "tA1", "tB1", "tBA1"]),
+        (False, ["ot2", "st2", "t2", "tA2", "tB2", "tBA2"]),
+    )
+    @ddt.unpack
+    def test_list_taxonomy_enabled_filter(self, enabled_parameter: bool, expected_taxonomies: list[str]) -> None:
+        """
+        Tests that the enabled filter works as expected
+        """
+        self._test_list_taxonomy(
+            user_attr="staff",
+            enabled_parameter=enabled_parameter,
+            expected_taxonomies=expected_taxonomies
+        )
+
+    @ddt.data(
+        ("orgA", ["st1", "st2", "t1", "t2", "tA1", "tA2", "tBA1", "tBA2"]),
+        ("orgB", ["st1", "st2", "t1", "t2", "tB1", "tB2", "tBA1", "tBA2"]),
+        ("orgX", ["st1", "st2", "t1", "t2"]),
+    )
+    @ddt.unpack
+    def test_list_taxonomy_org_filter(self, org_parameter: str, expected_taxonomies: list[str]) -> None:
+        """
+        Tests that the org filter works as expected
+        """
+        self._test_list_taxonomy(
+            user_attr="staff",
+            org_parameter=org_parameter,
+            expected_taxonomies=expected_taxonomies,
+        )
+
+    def test_list_taxonomy_invalid_org(self) -> None:
+        """
+        Tests that using an invalid org in the filter will raise BAD_REQUEST
+        """
         url = TAXONOMY_ORG_LIST_URL
 
-        self.client.force_authenticate(user=self.userS)
+        self.client.force_authenticate(user=self.staff)
 
-        # Set parameters cleaning empty values
         query_params = {"org": "invalidOrg"}
 
         response = self.client.get(url, query_params, format="json")
 
         assert response.status_code == status.HTTP_400_BAD_REQUEST
 
     @ddt.data(
-        ("user", ("tA1", "tB1", "tC1"), None),
-        ("userA", ("tA1", "tB1", "tC1"), None),
-        ("userS", ("st2", "t1", "t2"), "3"),
+        ("user", (), None),
+        ("staffA", ["tA2", "tBA1", "tBA2"], None),
+        ("staff", ["st2", "t1", "t2"], "3"),
     )
     @ddt.unpack
-    def test_list_taxonomy_pagination(self, user_attr, expected_taxonomies, expected_next_page):
+    def test_list_taxonomy_pagination(
+        self, user_attr: str, expected_taxonomies: list[str], expected_next_page: str | None
+    ) -> None:
+        """
+        Tests that the pagination works as expected
+        """
         url = TAXONOMY_ORG_LIST_URL
 
         if user_attr:
```

Comment:
nit: I aim for 100% coverage for test files, and coverage for test_views.py is really close: 96.6%

There's several places where you check `if user_attr`, but there's always a `user_attr` provided. You can see these lines marked out in yellow on the [coverage report](https://app.codecov.io/github/openedx/edx-platform/commit/32d20b2af85af0e7eafc033abc2161e4b28c0c22/blob/openedx/core/djangoapps/content_tagging/rest_api/v1/tests/test_views.py), and they can easily be covered like this:

```suggestion
        assert user_attr
        user = getattr(self, user_attr)
        self.client.force_authenticate(user=user)
```

Reply:
Fixed @pomegranited ! Thank you for catching that!